feat: add `stats/incr/nancovariance`

Adithya-jh opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Resolves #5554.

## Description

> What is the purpose of this pull request?

This pull request:

-  This pull request adds the nancovariance module, which computes an incremental sample covariance while ignoring input pairs that contain NaN values. 

## Related Issues

> Does this pull request have any related issues?


This pull request:

-   resolves #5554  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Adithya-jh committed:
feat(stats/incr): add nancovariance accumulator

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/incr/nancovariance/lib/index.js:
```diff
@@ -0,0 +1,52 @@
+/**
+* Compute an incremental sample covariance while ignoring NaN values.
+*
+* @module @stdlib/stats/incr/nancovariance
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/stats/incr/nancovariance/lib/index.js:
```diff
@@ -0,0 +1,52 @@
+* @module @stdlib/stats/incr/nancovariance
+*
+* @example
+* var incrnancount = require( '@stdlib/stats/incr/nancovariance' );
```
This example is incorrect.

kgryte commented on lib/node_modules/@stdlib/stats/incr/nancovariance/lib/main.js:
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable stdlib/jsdoc-no-space-aligned-asterisks */
```
Do NOT disable this lint rule. The lint rule is telling you explicitly our conventions.

kgryte commented on lib/node_modules/@stdlib/stats/incr/nancovariance/lib/main.js:
```diff
@@ -0,0 +1,87 @@
+// MAIN //
+
+/**
+ * Returns an accumulator function which incrementally computes a sample covariance.
```
This alignment is incorrect and does not follow our conventions.

kgryte commented on lib/node_modules/@stdlib/stats/incr/nancovariance/lib/main.js:
```diff
@@ -0,0 +1,87 @@
+ * // returns 0
+ */
+function nancovariance() {
+	var meanX = 0.0;
```
This implementation is not what we want. The implementation should be a thin wrapper around `incr/covariance`.

kgryte commented on lib/node_modules/@stdlib/stats/incr/nancovariance/package.json:
```diff
@@ -0,0 +1,60 @@
+{
+    "name": "@stdlib/stats/incr/nancovariance",
```
The indentation in this file is off. Should be 2-space indentation.

kgryte commented on lib/node_modules/@stdlib/stats/incr/nancovariance/test/test.js:
```diff
@@ -0,0 +1,53 @@
+	t.end();
+});
+
+// Additional tests...
```
In order for this PR to move forward, you need to add the additional tests. I recommend you simply reuse and modify the tests in `incr/covariance`.

kgryte reviewed (changes requested):
Left an initial round of comments.

kgryte commented:
You are also missing the following files:

- README.md
- docs/types/index.d.ts
- docs/types/test.ts

Adithya-jh commented:
Sure i will look into all the changes.
Thanks for reviewing.

Adithya-jh commented:
Hi @kgryte 
i have made all the changes mentioned.
could u review and tell , if there is anything missing !

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nancovariance/docs/types/index.d.ts:
```diff
@@ -0,0 +1,68 @@
+
+
+/**
+* Returns an accumulator function which incrementally computes a count, ignoring `NaN` values.
```
Says "computes a count" — should be "computes a sample covariance".

```suggestion
* Returns an accumulator function which incrementally computes a sample covariance, ignoring `NaN` values.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nancovariance/lib/main.js:
```diff
@@ -0,0 +1,85 @@
+// MAIN //
+
+/**
+* Returns an accumulator function which incrementally computes a sample covariance.
```
Missing "ignoring `NaN` values" in the JSDoc description.

```suggestion
* Returns an accumulator function which incrementally computes a sample covariance, ignoring `NaN` values.
```

Planeshifter reviewed (changes requested):
Some additional feedback.

Planeshifter committed:
fix: address review feedback

Correct the JSDoc and TypeScript descriptions to describe a sample
covariance ignoring NaN values, and align the TypeScript declarations
and tests with the `stats/incr/covariance` conventions (lowercase
`accumulator` type, relative import, and per-argument error checks).

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/nancovariance

Planeshifter committed:
chore: update copyright years

Planeshifter committed:
docs: fix REPL help to describe nancovariance accumulator

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
style: align spacing and use format for benchmark label

Planeshifter committed:
docs: move explanatory notes off the return-value annotations

The `// returns` annotation must state exactly the returned value; move
the explanatory parentheticals onto preceding comment lines so the REPL
help lint passes.

Planeshifter committed:
style: add blank line before test section separator

Match the two-blank-line convention preceding the `// TESTS //`
separator used across sibling packages.

Planeshifter committed:
style: use standard test section separator

Use the `// TESTS //` separator and drop the non-conventional
`// MAIN //` heading and `// Additional tests...` comment, matching
sibling packages such as stats/incr/covariance.

Planeshifter committed:
docs: remove hand-authored See Also entries

The related-packages section is autopopulated by tooling; empty the
`See Also` block and related-links so entries are not authored by hand.

Planeshifter committed:
docs: align description phrasing with sibling convention

Use "compute a sample covariance incrementally, ignoring NaN values"
across the package description, README, and module documentation,
matching the phrasing used by sibling stats/incr/nan* packages.

Planeshifter reviewed (approved):
Looks alright to me after clean-up.
